openshift/aos-cd-jobs
# Use repo files with proper authentication; install two missing packages

This directory (on jenkins) has files that can "patch" the aws e2c instance where things are run:

```
# ls -l /var/lib/jenkins/tweaks/
total 340
-rw-r--r--. 1 root root 285848 Feb  3 07:33 dnsmasq-2.76-16.el7.x86_64.rpm
-rw-r--r--. 1 root root   3179 Feb  3 07:33 good_repos_rhui.tgz  <-- does not contain creds
-rw-r--r--. 1 root root  53704 Feb  3 07:33 iptables-services-1.4.21-35.el7.x86_64.rpm
```

We can work on this more later (including generating the `/etc/yum.repod.d` files instead of just un'taring them there and adding the two .rpms to mirror.openshift.com). But for now, the ansible playbooks finish and get farther than they have in a while.

The `good_repos_rui.tgz` was created by taking the existing repos on the VM, updating  baseurl, removing credentials, and tar'ing up the files in place.

Test method:

* Make a PR [like this](https://github.com/openshift/origin-aggregated-logging/pull/2239)
* Make these modifications and push them to Jenkins for the two tests mentioned in the PR in the first step
```
$ ./sjb/generate.sh
$ ./sjb/push-update.sh sjb/generated/test_pull_request_origin_aggregated_logging_json_file-release-3.11.xml
$ ./sjb/push-update.sh sjb/generated/test_pull_request_origin_aggregated_logging_journald-release-3.11.xml
```
* Run the tests, check the logs

Here are logs:
  * https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_origin-aggregated-logging/2239/test_pull_request_origin_aggregated_logging_journald-release-3.11/849/build-log.txt
  * https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_origin-aggregated-logging/2239/test_pull_request_origin_aggregated_logging_json_file-release-3.11/647/build-log.txt
